### PR TITLE
Add Notice and DISCLAMER file

### PR DESCRIPTION
**Pull request by frankliee**
### What changes were proposed in this pull request?
Add a notice file.

References:
https://github.com/apache/inlong/blob/master/NOTICE
https://github.com/apache/spark/blob/master/NOTICE 
https://infra.apache.org/licensing-howto.html

### Why are the changes needed?
For apache release.
https://github.com/apache/incubator-uniffle/issues/207 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
No need.

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+The initial codebase was donated to the ASF by Tencent, copyright 2020-2022.
+
+--------------------------------------------------------------------------------
+
+Apache Spark
+Copyright 2014 and onwards The Apache Software Foundation.
+
+Apache Hadoop
+Copyright 2006 and onwards The Apache Software Foundation.
```
Comment (jerqi) [inline NOTICE:16]: Add blank line

Comment (frankliee) [reply NOTICE]: Added

**Comment by jerryshao** [comment]
Can you please check the contents of NOTICE from more projects, and the requirements from ASF?

**Comment by frankliee** [comment]
> Can you please check the contents of NOTICE from more projects, and the requirements from ASF?

For source release, it seems that we do not need extra contents.

`For any bundled dependency, consider whether LICENSE and/or NOTICE need to be modified. Do not modify LICENSE or NOTICE for non-bundled dependencies.`

https://infra.apache.org/licensing-howto.html

**Comment by jerryshao** [comment]
So maybe we don't need to list spark and hadoop license?

**Comment by frankliee** [comment]
> So maybe we don't need to list spark and hadoop license?

Yes, I have removed them in NOTICE.

File: NOTICE [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+Apache Uniffle
```
Comment (jerqi) [inline NOTICE:2]: Incubating?

Comment (frankliee) [reply NOTICE]: added

File: NOTICE [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+Apache Uniffle (incubating)
```
Comment (jerqi) [inline NOTICE:2]: First character should be upper case.

Comment (frankliee) [reply NOTICE:2]: No need for (word)
https://github.com/apache/incubator-mxnet/blob/master/NOTICE
https://github.com/apache/incubator-devlake/blob/main/NOTICE

Comment (jerqi) [reply NOTICE:2]: Consistent with DISCLAIMER.

Comment (frankliee) [reply NOTICE:2]: Both are updated

File: NOTICE [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+Copyright 2022 and onwards The Apache Software Foundation.
+
+This product includes software developed at
+The Apache Software Foundation (https://www.apache.org/).
+
+The initial codebase was donated to the ASF by Tencent, copyright 2020-2022.
+
```
Comment (jerryshao) [inline NOTICE:8]: Additional blank line.

Comment (frankliee) [reply NOTICE]: fixed

**Review by jerryshao — APPROVED** [review]

**Pull request merged**